integration/spark/src/main/scala/org/apache/spark/sql/execution/command/index/CarbonRefreshIndexCommand.scala

```diff
@@ -71,30 +80,81 @@ case class CarbonRefreshIndexCommand(
   private def refreshIndex(
       sparkSession: SparkSession,
       parentTable: CarbonTable,
-      indexOp: Optional[DataMapSchema]): Unit = {
-    val schema = indexOp.get
+      indexMetaData: IndexMetadata): Unit = {
+    var indexInfo: util.Map[String, String] = new util.HashMap[String, String]()
+    val allIndexesIterator = indexMetaData.getIndexesMap.entrySet().iterator()
+    breakable {
+      while (allIndexesIterator.hasNext) {
+        val currentIndex = allIndexesIterator.next()
+        if (!currentIndex.getKey.equalsIgnoreCase(CarbonIndexProvider.SI.getIndexProviderName)) {
+          val indexIterator = currentIndex.getValue.entrySet().iterator()
+          while (indexIterator.hasNext) {
+            val indexEntry = indexIterator.next()
+            if (indexEntry.getKey.equalsIgnoreCase(indexName)) {
```

Comment:
it's map. why are you looping every element. Just check containsKey() no need of iterator here

Reply:
done